packages/core/src/extension/extension-messenger.ts

```diff
@@ -0,0 +1,111 @@
+import CheckoutStore from '../checkout/checkout-store';
+import { IframeEventListener, IframeEventPoster } from '../common/iframe';
+
+import { ExtensionNotFoundError } from './errors';
+import { UnsupportedExtensionCommandError } from './errors/unsupported-extension-command-error';
+import { Extension } from './extension';
+import { ExtensionCommandHandler } from './extension-command-handler';
+import { ExtensionCommand, ExtensionOriginEventMap } from './extension-origin-event';
+import { HostOriginEvent } from './host-origin-event';
+
+export class ExtensionMessenger {
+    private _extensions: Extension[] | undefined;
+
+    constructor(
+        private _store: CheckoutStore,
```

Comment:
Preferably use the read-only version, i.e.: `ReadableCheckoutStore`

Reply:
Yes, `ReadableCheckoutStore` is used in ways I was not aware of before.